pkgs/development/python-modules/disnake/default.nix

```diff
@@ -0,0 +1,63 @@
+{ lib
+, aiohttp
+, buildPythonPackage
+, fetchFromGitHub
+, libopus
+, pynacl
+, pythonOlder
+, withVoice ? true
+, ffmpeg
+}:
+
+buildPythonPackage rec {
+  pname = "disnake";
+  version = "2.9.2";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "DisnakeDev";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-J7nNVHhkQ9ekGjioObxaTL0OhjLjHHYlgbPdAG9Vz4o=";
+  };
+
+  propagatedBuildInputs = [
+    aiohttp
+  ] ++ lib.optionals withVoice [
+    libopus
+    pynacl
+    ffmpeg
+  ];
```

Comment:
Please use `optional-dependencies` instead.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#how-to-add-optional-dependencies-python-optional-dependencies

Reply:
The problem with using optional dependencies like that is that the package itself needs to apply certain patches in order for the optional dependencies to be used properly. Would the package need to have both a `withVoice` toggle and have downstream packages add in the optional dependencies, even though the optional dependencies are included by default?

Reply:
Actually now that I think about it, it's kinda weird that `withVoice` is on by default. I wonder why `discord.py` has it on by default, since it's in inherited from that. Nonetheless, the need to patch files with paths of optional dependencies muddies this a bit.